identity/ideth/service.go

```diff
@@ -317,6 +318,21 @@ func (i service) GetKeysByPurpose(did id.DID, purpose *big.Int) ([][32]byte, err
 
 }
 
+// CurrentP2PKey returns the latest P2P key
+func (i service) CurrentP2PKey(did id.DID) (ret string, err error) {
+	keys, err := i.GetKeysByPurpose(did, big.NewInt(id.KeyPurposeP2P))
+	if err != nil {
+		return ret, err
+	}
+	lastKey := keys[len(keys)-1]
+	p2pID, err := ed25519.PublicKeyToP2PKey(lastKey)
```

Comment:
The method is similar to `GetClientP2PURL` I think we could combine some parts after the migration.

Maybe we could add the following check. 
```golang
if key.RevokedAt.Cmp(big.NewInt(0)) != 0 {
       return "", errors.New("p2p key has been revoked")
}
```